### PR TITLE
Add SSL Breaking changes for logstash-output-http to breaking changes…

### PR DESCRIPTION
**Pull request by robbavey**
… doc

**PREVIEW:** https://logstash_bk_16701.docs-preview.app.elstc.co/guide/en/logstash/master/breaking-changes.html

### CODE REVIEW

**Comment by donoghuc** [comment]
I like the idea here. For all the other plugins we will continue with sections that have the plugin name and the obsolete parameters. Looks like the docs build is red. The only thing I saw in the logs is a warning about use of headers. Not sure if that is the cause. Once we get this base example sorted out we can start making sure the other plugins get a similar treatment :)

File: docs/static/breaking-changes-90.asciidoc [hunk 1]
```diff
@@ -13,3 +13,26 @@ Include:
...
+| cacert |<<plugins-outputs-http-ssl_certificate_authorities>>
+| client_cert |<<plugins-outputs-http-ssl_certificate>>
+| client_key |<<plugins-outputs-http-ssl_key>>
+| keystore |<<plugins-outputs-http-ssl_keystore_path>>
+| keystore_password |<<plugins-outputs-http-ssl_keystore_password>>
+| keystore_type |<<plugins-outputs-http-ssl_keystore_password>>
+| truststore |<<plugins-outputs-http-ssl_truststore_path>>
+| truststore_password |<<plugins-outputs-http-ssl_truststore_password>>
+| truststore_type |<<plugins-outputs-http-ssl_truststore_type>>
+|=======================================================================
```
Comment (karenzone) [inline docs/static/breaking-changes-90.asciidoc:38]:
```suggestion
[discrete]
[[ssl-settings-9.0]]
===== Changes to SSL settings in {ls} plugins

We've removed deprecated SSL settings in some {ls} plugins, and have replaced them with updated settings.
If your plugin configuration contains any of these obsolete options, the plugin may fail to start.

**Plugins with changes to SSL settings** 

[discrete]
[[output-http-ssl-9.0]]
.<<plugins-outputs-http,logstash-output-http>>

[%collapsible]
====

[cols="<,<",options="header",]
|=======================================================================
|Setting|Replaced by
| cacert |<<plugins-outputs-http-ssl_certificate_authorities>>
| client_cert |<<plugins-outputs-http-ssl_certificate>>
| client_key |<<plugins-outputs-http-ssl_key>>
| keystore |<<plugins-outputs-http-ssl_keystore_path>>
| keystore_password |<<plugins-outputs-http-ssl_keystore_password>>
| keystore_type |<<plugins-outputs-http-ssl_keystore_password>>
| truststore |<<plugins-outputs-http-ssl_truststore_path>>
| truststore_password |<<plugins-outputs-http-ssl_truststore_password>>
| truststore_type |<<plugins-outputs-http-ssl_truststore_type>>
|=======================================================================

====
```

Comment (karenzone) [reply docs/static/breaking-changes-90.asciidoc:38]: Let's consider setting this up as expandable/collapsible sections to help users focus on the plugins they're most interested in.

Screenshot 2024-11-21 at 12 02 53 PM

Comment (karenzone) [reply docs/static/breaking-changes-90.asciidoc:38]: Other general guidelines:
* Make heading anchors more specific (examples in lines 17 and 26).
* Be sure to describe user impact (line 21). 
* Include link to related docs if appropriate (line 27)
* Follow Elastic docs standard for sentence style headings (line 18).

**Review by karenzone — COMMENTED** [review]
Left suggestions and broader guidelines inline. 
Let's discuss and iterate. When we agree, let's appy these style guidelines for the other 10 plugins with changes to SSL settings.

File: docs/static/breaking-changes-90.asciidoc (hunk 1 above)
Comment (robbavey) [reply docs/static/breaking-changes-90.asciidoc:38]: Big fan of the collapsible sections - there's going to be a lot of similar content here, so anything that conveys everything we need without dominating the whole breaking changes docs works for me.

Other suggestions all make sense to me too

**Comment by robbavey** [comment]
Doc Preview:

https://logstash_bk_16701.docs-preview.app.elstc.co/

**Comment by karenzone** [comment]
### Follow up: 
Now that I've played with it, I still like the collapsible content, but I don't like the interaction details in my suggestion. 
The down arrow and the link to plugin docs are too close together, and make for an ambiguous information experience. Restating the obvious, if you click the down arrow, the section expands to show deleted settings. If you click the plugin text, you jump to the plugin docs. I'll tweak my suggestion accordingly. 

Also, the dropdown arrow is a common UI element and we use it extensively in our docs. I'll put in a note to explain, and we can use it--or not. 

Screenshot 2024-11-22 at 3 28 22 PM

File: docs/static/breaking-changes-90.asciidoc [hunk 2]
```diff
@@ -13,3 +13,34 @@ Include:
 * user impact/decription/value prop
 * link to relevant docs for more information
 
+[discrete]
+[[ssl-settings-9.0]]
+===== Changes to SSL settings in {ls} plugins
+
+We've removed deprecated SSL settings in some {ls} plugins, and have replaced them with updated settings.
+If your plugin configuration contains any of these obsolete options, the plugin may fail to start.
+
```
Comment (karenzone) [inline docs/static/breaking-changes-90.asciidoc:22]:
```suggestion

Click the arrow beside a plugin name to see the list of settings that have been
removed and their replacements. 

```

File: docs/static/breaking-changes-90.asciidoc [hunk 3]
```diff
@@ -13,3 +13,34 @@ Include:
...
+**Plugins with changes to SSL settings** 
+
+[discrete]
+[[output-http-ssl-9.0]]
+.<<plugins-outputs-http,logstash-output-http>>
```
Comment (karenzone) [inline docs/static/breaking-changes-90.asciidoc:27]:
```suggestion
.`logstash-output-http`
```

File: docs/static/breaking-changes-90.asciidoc (hunk 2 above)
Comment (karenzone) [reply docs/static/breaking-changes-90.asciidoc:22]: The expandable content is a common UI element, and we use it extensively throughout our docs.  If you think we need to call it it, here's a suggestion for how and where to do that.

File: docs/static/breaking-changes-90.asciidoc (hunk 3 above)
Comment (karenzone) [reply docs/static/breaking-changes-90.asciidoc:27]: The link to the plugin docs added confusion and seems like overkill.  All of the links in the dropdown content take users to the plugin docs anyway.

File: docs/static/breaking-changes-90.asciidoc (hunk 2 above)
Comment (robbavey) [reply docs/static/breaking-changes-90.asciidoc:22]: I think I'm good with skipping it. If there is plenty of prior art of us using the control, and not having any explanatory text, then I'm good

**Review by karenzone — APPROVED** [review]
LGTM!

**Pull request merged**